### PR TITLE
Add JavaScript snippets

### PR DESCRIPTION
**Pull request by Axorax**
# Description

## Type of Change

- [x] ✨ New snippet
- [ ] 🛠 Improvement to an existing snippet
- [ ] 🐞 Bug fix
- [ ] 📖 Documentation update
- [ ] 🔧 Other (please describe):

## Checklist

- [x] I have tested my code and verified it works as expected.
- [x] My code follows the style and contribution guidelines of this project.
- [x] Comments are added where necessary for clarity.
- [x] Documentation has been updated (if applicable).
- [x] There are no new warnings or errors from my changes.

## Related Issues

Closes #

## Screenshots (Optional)

<details> 
<summary>Click to view screenshots</summary> 

 

</details>

### CODE REVIEW

**Comment by Axorax** [comment]
closed previous one due to merge conflicts. I also removed the repeated ones and resolved conflicts.

It's a bit hard to see what's already added because the site search doesn't work.

File: public/data/javascript.json [hunk 1]
```diff
@@ -822,6 +822,133 @@
...
+          "",
+          "// Usage:",
+          "const fetchData = () => console.log('Fetching data...');",
+          "const rateLimitedFetch = rateLimit(fetchData, 2, 1000);",
+          "setInterval(() => rateLimitedFetch(), 200); // Only calls fetchData twice every second"
+        ],
+        "tags": ["javascript", "function", "rate-limiting", "utility"],
+        "author": "axorax"
+      },
+      {
```
Comment (Mathys-Gasnier) [inline public/data/javascript.json:940]: That shouldn't be here

File: public/data/javascript.json [hunk 2]
```diff
@@ -906,6 +1084,38 @@
...
+          "};",
+          "",
+          "// Usage:",
+          "console.log(isItemInLocalStorage('user')); // Output: true or false"
+        ],
+        "tags": ["javascript", "localStorage", "storage", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Check bytes used",
```
Comment (Mathys-Gasnier) [inline public/data/javascript.json:1103]: I still find this one weird, it doesn't return the quantity of bytes used, IMO just remove it, there is no way in JS of knowing the length in bytes of a type, and the localstorage length can already be accessed

**Review by Mathys-Gasnier — CHANGES_REQUESTED** [review]

**Comment by Mathys-Gasnier** [comment]
> closed previous one due to merge conflicts. I also removed the repeated ones and resolved conflicts.
> 
> It's a bit hard to see what's already added because the site search doesn't work.

You don't need to close, and then reopen to resolve merge conflicts, you can merge main into your fork/branch or use the tools github provides.
Please avoid closing and re openning issues like this, it makes reviewing a pain.

**Comment by Axorax** [comment]
> > closed previous one due to merge conflicts. I also removed the repeated ones and resolved conflicts.
> > It's a bit hard to see what's already added because the site search doesn't work.
> 
> You don't need to close, and then reopen to resolve merge conflicts, you can merge main into your fork/branch or use the tools github provides. Please avoid closing and re openning issues like this, it makes reviewing a pain.

Sorry. I didn't close it myself. I went to fix the conflict and the issue closed by itself.

**Comment by Axorax** [comment]
hopefully it's all good now

File: public/data/javascript.json [hunk 3]
```diff
@@ -910,33 +910,6 @@
         "tags": ["javascript", "function", "compose", "utility"],
         "author": "axorax"
       },
-      {
-        "title": "Rate Limit Function",
```
Comment (Mathys-Gasnier) [inline public/data/javascript.json]: The `Random string` is a duplicate, this one was fine... Sorry i should have included the name of the snippet

**Review by Mathys-Gasnier — CHANGES_REQUESTED** [review]

**Comment by Axorax** [comment]
Done!

**Review by Mathys-Gasnier — APPROVED** [review]

**Pull request merged**